features/facebookImport/src/context/importer-context.jsx

```diff
@@ -125,12 +134,24 @@ export const ImporterProvider = ({ children }) => {
         return storage.removeFile(fileID);
     };
 
+    const handleSelectFile = async () => {
+        const { polyNav } = pod;
+        setFiles(null); // To show the loading overlay
+        try {
+            setSelectedFileUrl(await polyNav.pickFile("application/zip"));
+        } catch (error) {
+            setGlobalError(new FileSelectionError(error));
+        }
+        refreshFiles();
+    };
+
     const handleImportFile = async () => {
-        const { polyNav, polyOut } = pod;
+        if (!selectedFileUrl) return;
+        const { polyOut } = pod;
         setFiles(null); // To show the loading overlay
         try {
-            const url = await polyNav.pickFile("application/zip");
-            if (url) await polyOut.importArchive(url);
+            await polyOut.importArchive(selectedFileUrl);
+            setSelectedFileUrl(null);
```

Comment:
Again, some action at a distance going on here... I would be more comfortable if this was an object such as selectedFileUrl set to "imported" or something like that. I guess that's to avoid handleImportFile from selecting twice, but couldn't we store state in some more semantic way? I see this is React, but still...

Reply:
I think it'd be cleaner to actually expose state (as in state machine) using an enum, state pattern or similar. But I do think that's too big for feature freeze.